src/backend/d3d12/FramebufferD3D12.cpp

```diff
@@ -41,31 +40,26 @@ namespace d3d12 {
                 device->GetD3D12Device()->CreateRenderTargetView(texture.Get(), &rtvDesc, rtvHandle);
             } else {
                 // TODO(enga@google.com) no attachment. This will use the backbuffer. Remove when this hack is removed
-                usingBackbuffer = true;
                 rtvIndex++;
             }
         }
 
-        // TODO(enga@google.com): load depth attachment from subpass
-        if (usingBackbuffer) {
+        if (info.depthStencilAttachmentSet) {
             dsvHeap = device->GetDescriptorHeapAllocator()->AllocateCPUHeap(D3D12_DESCRIPTOR_HEAP_TYPE_DSV, 1);
-            D3D12_CPU_DESCRIPTOR_HANDLE dsvHandle = dsvHeap.GetCPUHandle(0);
 
-            D3D12_DEPTH_STENCIL_VIEW_DESC dsvDesc;
-            dsvDesc.Format = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
-            dsvDesc.ViewDimension = D3D12_DSV_DIMENSION_TEXTURE2D;
-            dsvDesc.Texture2D.MipSlice = 0;
-            dsvDesc.Flags = D3D12_DSV_FLAG_NONE;
+            uint32_t attachment = info.depthAttachment;
+            auto textureView = currentFramebuffer->GetTextureView(attachment);
+            ComPtr<ID3D12Resource> texture = ToBackend(textureView->GetTexture())->GetD3D12Resource();
+            D3D12_CPU_DESCRIPTOR_HANDLE dsvHandle = dsvHeap.GetCPUHandle(0);
```

Comment:
And then this will need to be something like `dsvHeap.GetCPUHandle(depthStencilAttachmentIndex);`